source/adios2/helper/adiosMemory.inl

```diff
@@ -514,7 +514,7 @@ void Resize(std::vector<T> &vec, const size_t dataSize, const bool debugMode,
 // functions) and copies to the output buffer in blocks. the memory address
 // calculation complexity for copying each block is minimized to O(1), which is
 // independent of the number of dimensions.
-static void NdCopyRecurDFSeqPadding(size_t curDim, const char *&inOvlpBase,
+static inline void NdCopyRecurDFSeqPadding(size_t curDim, const char *&inOvlpBase,
```

Comment:
I doubt explicitly marking these inline actually makes any difference since it's non-binding and just a "suggestion" to the compiler anyways.  The likelihood is pretty high that these were already getting inlined by compilers.  Regardless though, it's good for the sake of complete to mark them as such, so I'm supportive of this.